### ISSUE TITLE
Module not found: Error: Can't resolve 'hiredis'

### ISSUE THREAD

**Issue by richburdon** [liady/webpack-node-externals#34]
I use nodeExternals a lot to build server-side modules, but I get this error for any project that depends (indirectly) on redis (all other modules -- 1000s of them -- work fine).

In this case, my import of `bull` causes problems, through its dependency on `ioredis` (similar case here https://github.com/webpack/webpack-dev-server/issues/227)

```
ERROR in ../sub/scheduler/~/bull/~/redis-parser/lib/hiredis.js
Module not found: Error: Can't resolve 'hiredis' in '/.../scheduler/node_modules/bull/node_modules/redis-parser/lib'
 @ ../scheduler/~/bull/~/redis-parser/lib/hiredis.js 3:14-32
 @ ../scheduler/~/bull/~/redis-parser/lib/parser.js
 @ ../scheduler/~/bull/~/redis-parser/index.js
 @ ../scheduler/~/bull/~/ioredis/lib/redis/parser.js
 @ ../scheduler/~/bull/~/ioredis/lib/redis.js
 @ ../scheduler/~/bull/~/ioredis/index.js
 @ ../scheduler/~/bull/lib/queue.js
 @ ../scheduler/src/util/queue.js
 @ ../scheduler/index.js
 @ ./src/server/router/admin.js
 @ ./src/server/server.js
 @ ./src/server/main.js
 @ multi babel-polyfill ./src/server/main.js
```

**Comment by muratgozel** [comment]
I have a simple webpack configuration for server side bundling and faced an issue exactly like yours. Adding nodeExternals() plugin solved the problem in my case. Here is related code in my webpack configuration:

```javascript
{
  ...
  externals: [nodeExternals()],
  target: 'node',
  ...
}
```

It just skips all packages inside node_modules directory while bundling.

**Comment by sprinklr-gurgaon** [comment]
But why do you skip all packages inside node_modules while bundling?
This looks like a workaround and not a proper fix?

**Comment by muratgozel** [comment]
Yes, it is not a proper fix but we don't need bundling in server side anyway. Not just packages inside node_modules folder, any package or module we import our code may not have to be bundled in server side since node supports es6 modules. Maybe redis lib maintainer clarify this problem better.

**Comment by anilanar** [comment]
@sprinklr-gurgaon Because most node-only libraries do lazy `require`ing using `require`, which causes webpack to fail.
